Mage/src/main/java/mage/abilities/effects/common/continuous/PlayTheTopCardEffect.java

```diff
@@ -47,12 +50,27 @@ public PlayTheTopCardEffect copy() {
 
     @Override
     public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        return applies(objectId, null, source, game, affectedControllerId);
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability affectedAbility, Ability source, Game game, UUID playerId) {
         Card cardOnTop = game.getCard(objectId);
+        Card cardToCheckProperties = cardOnTop;
+
+        // Check each ability individually, as e.g. Adventures and associated creatures may get different results from the filter.
+        if (affectedAbility != null) {
+            MageObject sourceObject = affectedAbility.getSourceObject(game);
+            if (sourceObject != null && sourceObject instanceof Card) {
+                cardToCheckProperties = (Card) sourceObject;
+            }
+        }
```

Comment:
Same changes must be done for all other `play the top card` cards with same code like `Bolass Citadel`

Reply:
It won't matter for any play the top card cards except those where only some cards are playable (Precognition Field) or where it matters which card is cast (Bolas's Citadel). But I'll take a look through as many as I can.